lib/galaxy/jobs/runners/local.py

```diff
@@ -134,9 +134,11 @@ def queue_job( self, job_wrapper ):
     def stop_job( self, job ):
         #if our local job has JobExternalOutputMetadata associated, then our primary job has to have already finished
         job_ext_output_metadata = job.get_external_output_metadata()
-        if job_ext_output_metadata:
+        try:
             pid = job_ext_output_metadata[0].job_runner_external_pid  # every JobExternalOutputMetadata has a pid set, we just need to take from one of them
-        else:
+            assert pid not in [ None, '' ]
+        except:
```

Comment:
I don't love the [bare](https://docs.python.org/2/howto/doanddont.html#except) ``except`` here - any chance you make this an ``except Exception:`` instead - that would still catch the ``AssertionError``.